freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/model/UsersStateDifference.java

```diff
@@ -183,4 +204,28 @@ public static UsersStateDifference forDeletedUser(String deletedUser, Collection
 
         return ImmutableMultimap.copyOf(groupMembershipToRemove);
     }
+
+    private static ImmutableSet<String> getAllUsers(UmsUsersState umsState) {
+        return umsState.getUsersState().getUsers().stream()
+                .map(FmsUser::getName)
+                .filter(username -> !FreeIpaChecks.IPA_PROTECTED_USERS.contains(username))
+                .collect(ImmutableSet.toImmutableSet());
+    }
+
+    private static ImmutableSet<String> getUsersWithStaleCredentials(UmsUsersState umsState, UsersState ipaState) {
+        return umsState.getUsersState().getUsers().stream()
+                .map(FmsUser::getName)
+                .filter(username -> !FreeIpaChecks.IPA_PROTECTED_USERS.contains(username) && credentialsAreStale(username, umsState, ipaState))
+                .collect(ImmutableSet.toImmutableSet());
+    }
+
+    private static boolean credentialsAreStale(String username, UmsUsersState umsState, UsersState ipaState) {
+        boolean updateCredentials = true;
+        UserMetadata ipaUserMetadata = ipaState.getUserMetadataMap().get(username);
+        if (ipaUserMetadata != null) {
+            WorkloadCredential umsCredential = umsState.getUsersWorkloadCredentialMap().get(username);
+            updateCredentials = ipaUserMetadata.getWorkloadCredentialsVersion() < umsCredential.getVersion();
```

Comment:
just use return here, no need for the variable. if you put the next line into an else, it would be obvious by looking at the method it has 2 exit point and no need to check where the variable could be modified during execution

Reply:
Done.